src/Microsoft.AspNet.Diagnostics/StatusCodePagesMiddleware.cs

```diff
@@ -25,8 +25,16 @@ public StatusCodePagesMiddleware(RequestDelegate next, [NotNull]StatusCodePagesO
 
         public async Task Invoke(HttpContext context)
         {
+            context.SetFeature<IStatusCodePagesFeature>(new StatusCodePagesFeature());
+
             await _next(context);
 
+            if (context.GetFeature<IStatusCodePagesFeature>() == null)
+            {
+                // No-op if the feature is turned off or missing.
```

Comment:
Recommended new comment :

    // Check if the feature is still available because other middleware (such as a web API written in MVC) could
    // have removed the feature to prevent HTML status code responses from showing up to an API client.